Implemented GetLatestRelease API.

chenjiaming93 opened the pull request:
trying to fix
#966

shiftkey commented on Octokit/Helpers/ApiUrls.cs:
```diff
@@ -140,6 +140,17 @@ public static Uri Releases(string owner, string name, int id)
+        /// <param name="owner">The owner of the repository</param>
+        /// <param name="name">The name of the repository</param>
+        /// <returns></returns>
+        public static Uri LatestReleases(string owner, string name)
```
:lipstick: perhaps make this singular i.e. `LatestRelease`

shiftkey commented:
Looking good so far @chenjiaming93! 

A couple of things I'd like to verify as part of integration tests:

 - that we're able to deserialize a release correctly using this endpoint
 - what happens when no releases exist for a project?

chenjiaming93 commented:
I have added tests. This one is similar to GetRelease API. Just replaced release id with "latest".  In the tests for getrelease we are only checking if we can receive the get request at that endpoint and throw proper exceptions when missing arguments. If there is no latest release, are we expected to get a Null JSON or we a  500 error?

shiftkey commented:
@chenjiaming93 we have some existing tests which use the real GitHub API - let's add a couple of tests for this endpoint there: https://github.com/octokit/octokit.net/blob/master/Octokit.Tests.Integration/Clients/ReleasesClientTests.cs

A couple of tests I think we can do to complete this feature:

 - "get the latest release for Octokit" - we get a non-null `Release` back
 - "get the latest release for a new repository" -> get a helpful error back

>  If there is no latest release, are we expected to get a Null JSON or we a 500 error?

I'm honestly not sure - let's find out by writing the test!

I think the end result should be a custom exception that we throw to the user. There's a detailed example [here](https://github.com/octokit/octokit.net/blob/f0111aa72239da06c6268ae0c47562957f6603bc/Octokit/Clients/RepositoriesClient.cs#L75-L127) for how we handle when creating a repository fails - I think you'll have to do something similar and `try-catch` the call and then check if the error is something we're familiar with.

chenjiaming93 commented:
@shiftkey  I see. I will add integration tests soon.

khalidabuhakmeh commented:
Any update on this issue?

chenjiaming93 commented:
@khalidabuhakmeh  I am busy with my final exams this week. I will add integration tests this weekend.

Haacked commented:
Happy New Year! @chenjiaming93 were you still planning to add integration tests?

chenjiaming93 commented:
@Haacked  I'm just back to school. Will start working tmr

shiftkey commented:
I've moved the integration tests out to #1086 rather than hold this up any longer.

Thanks for the contribution @chenjiaming93!
